pkg/standalone-utils/test/LidoRelayer.test.ts

```diff
@@ -1,462 +1,950 @@
-          sharedBeforeEach('build swap request', async () => {
-            singleSwap = {
-              kind: SwapKind.GivenIn,
-              poolId: basePoolId,
-              assetIn: basePoolTokens.findBySymbol('wstETH').address,
-              assetOut: basePoolTokens.WETH.address,
-              amount: fp(1),
-              userData: '0x',
-            };
-          });
+      function testWrap(): void {
+        it('wraps with immediate amounts', async () => {
+          const amount = fp(1);
+          const expectedWstETHAmount = await wstETH.instance.getWstETHByStETH(amount);
 
-          it('performs the given swap', async () => {
-            const receipt = await relayer.connect(sender).swap(singleSwap, funds, limit, deadline);
+          const receipt = await (
+            await relayer.connect(sender).multicall([encodeWrap(tokenSender, tokenRecipient, amount)])
+          ).wait();
 
-            expectEvent.inIndirectReceipt(await receipt.wait(), vault.instance.interface, 'Swap', {
-              poolId: singleSwap.poolId,
-              tokenIn: singleSwap.assetIn,
-              tokenOut: singleSwap.assetOut,
-              amountIn: singleSwap.amount,
-              // amountOut
-            });
-          });
+          const relayerIsRecipient = TypesConverter.toAddress(tokenRecipient) === relayer.address;
+          expectTransferEvent(
+            receipt,
+            {
+              from: relayerIsRecipient ? ZERO_ADDRESS : relayer.address,
+              to: relayerIsRecipient ? relayer.address : TypesConverter.toAddress(tokenRecipient),
+              value: expectedWstETHAmount,
+            },
+            wstETH
+          );
+        });
 
-          it('does not leave dust on the relayer', async () => {
-            await relayer.connect(sender).swap(singleSwap, funds, limit, deadline);
+        it('stores wrap output as chained reference', async () => {
+          const amount = fp(1);
+          const expectedWstETHAmount = await wstETH.instance.getWstETHByStETH(amount);
 
-            expect(await WETH.balanceOf(relayer)).to.be.eq(0);
-            expect(await wstETH.balanceOf(relayer)).to.be.eq(0);
-          });
+          const receipt = await (
+            await relayer
+              .connect(sender)
+              .multicall([encodeWrap(tokenSender, tokenRecipient, amount, toChainedReference(0))])
+          ).wait();
+
+          const relayerIsRecipient = TypesConverter.toAddress(tokenRecipient) === relayer.address;
+          const {
+            args: { value: wstETHAmount },
+          } = expectTransferEvent(
+            receipt,
+            {
+              from: relayerIsRecipient ? ZERO_ADDRESS : relayer.address,
+              to: relayerIsRecipient ? relayer.address : TypesConverter.toAddress(tokenRecipient),
+              value: expectedWstETHAmount,
+            },
+            wstETH
+          );
+          await expectChainedReferenceContents(toChainedReference(0), wstETHAmount);
```

Comment:
```suggestion
          await expectChainedReferenceContents(toChainedReference(0), expectedWstETHAmount);
```

We don't need this: we already know `value` (we set it to `expectedWstETHAmount`). And we've already tested for the transfer event in the previous test.